openwrt_api.py

```diff
@@ -0,0 +1,400 @@
+  while True:
+    recv_sock = socket.socket(socket.AF_INET, socket.SOCK_RAW,
+      socket.IPPROTO_ICMP)
+    send_sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM,
+      socket.IPPROTO_UDP)
+
+    send_sock.setsockopt(socket.SOL_IP, socket.IP_TTL, ttl)
+    recv_sock.bind(("", port))
+    bytesent = send_sock.sendto("", (dest_ip, port))
+
+    # Account for the resources
+    if _is_loopback_ipaddr(dest_ip):
+      nanny.tattle_quantity('loopsend', bytessent + 28)
+    else:
+      nanny.tattle_quantity('netsend', bytessent + 28)
+
+    curr_addr = None
+    curr_name = None
+
+    try:
+      # socket.recvfrom() gives back (data, address)
+      recv_sock.settimeout(waittime)
+      curr_addr = recv_sock.recvfrom(512)[1]
+      curr_addr = curr_addr[0]
+      try:
+        curr_name = socket.gethostbyaddr(curr_addr)[0]
+      except socket.error:
+        curr_name = curr_addr
+    except socket.error:
+      pass
+
+    send_sock.close()
+    recv_sock.close()
+
+    if curr_addr is not None:
+      curr_host = "%s (%s)" % (curr_name, curr_addr)
+    else:
+      curr_host = "*"
+    result.append("%d %s" % (ttl, curr_host)) 
+
+    ttl += 1
+    if curr_addr == dest_addr or ttl > max_hops:
+      break
+
+  return result
+
+def ping(dest_ip,count):
+  """
+  <Purpose>
+    Send ICMP ECHO_REQUEST to network hosts.
+
+  <Arguments>
+    dest_ip: the ip address to communicate with.
+    count: Stop after sending count ECHO_REQUEST packets.
+
+  <Exceptions>
+    RepyArgumentError when the host name is not valid types
+        or values
+
+  <Side Effects>
+    None
+
+  <Returns>
+    ping statistics as a dict, such as {'received': '2', 'jitter': '12.441', 
+    'avgping': '27.302', 'minping': '14.861', 'host': 'www.google.com', 
+    'maxping': '39.743', 'sent': '2'}
+    `host`: the target hostname that was pinged
+    `sent`: the number of ping request packets sent
+    `received`: the number of ping reply packets received
+    `minping`: the minimum (fastest) round trip ping request/reply
+        time in milliseconds
+    `avgping`: the average round trip ping time in milliseconds
+    `maxping`: the maximum (slowest) round trip ping time in
+        milliseconds
+    `jitter`: the standard deviation between round trip ping times
+        in milliseconds
+  """
+  if type(dest_ip) is not str:
+    raise RepyArgumentError("Provided host must be a string!")
+
+  if type(count) is not int:
+    raise RepyArgumentError("Provided count must be an int!")
+
+  # Check the input arguments (sanity)
+  if not emulcomm._is_valid_ip_address(dest_ip):
+    raise RepyArgumentError("Provided dest_ip is not valid! IP: '" + dest_ip + "'")
+
+  if count < 1:
+    raise RepyArgumentError("Provided count must be more than 0")
+
+  process = portable_popen.Popen(['ping', '-c', str(count), dest_ip],)
+  ping_output, err = process.communicate()
+  return _parse(str(ping_output))
+
+def get_station(interface):
+  """
```

Comment:
This should respect `emulcomm`'s / the nodemanager's configured allowed interfaces list.